feat: add connection to mongodb using mongoose

permil-successive opened the pull request:
use of promise

@Prabal555 @nxvinay

permil-successive committed:
feat: 🎸 add validation handler (middleware)

add validation handler (middleware) which validate input variables in
the request as per config file.

✅ Closes: #39524

permil-successive committed:
fix: 🐛 fix throw error on unwanted variable in required

permil-successive committed:
feat: 🎸 add auth middleware (Authentication & Authorization)

✅ Closes: #40009

permil-successive committed:
refactor: 💡 beautify the code

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009

permil-successive committed:
refactor: 💡 add comments and break longer functions

add comments, break longer functions, merge import/export in single
statement wherever required

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009

permil-successive committed:
fix: 🐛 remove extra supplied variable

permil-successive committed:
fix: 🐛 fix regex bug

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009

permil-successive committed:
feat: 🎸 add conection to mongodb using mongoose and promise

✅ Closes: #40010

Prabal555 commented on package.json:
```diff
@@ -45,8 +45,11 @@
   "dependencies": {
     "@types/body-parser": "^1.17.1",
     "@types/express": "^4.17.2",
+    "@types/mongoose": "^5.7.0",
```
good practice to have @types in devDependencies

Prabal555 commented on .env:
```diff
@@ -1,2 +1,4 @@
 PORT=9000
 ENV=dev
+SECRET_KEY=qwertyuiopasdfghjklzxcvbnm123456
+MONGO_URL=mongodb://localhost:27017/express-training
```
don't commit .env use a .env.example instead

Prabal555 commented on src/libs/Database.ts:
```diff
@@ -0,0 +1,22 @@
+
+  close(): Promise<any> {
+    return mongoose.disconnect();
+  }
```
can use async await instead of .then .catch

permil-successive committed:
refactor: 💡 do changes as request in PR#10

remove unwanted route from server.ts, add a single response helper, fix
trainee delete route, fix custom validation function of trainee config,
fix response status code, use try/catch and pure function in
validationHandler.ts

✅ Closes: #39524

permil-successive committed:
fix: 🐛 fix bug in validation handler

for string datatype

permil-successive committed:
Merge branch 'feature/39524' of https://github.com/permil-st/javascript-server into feature/40009 and changes request in PR#11

permil-successive committed:
Merge branch 'feature/40009' of https://github.com/permil-st/javascript-server into feature/40010

permil-successive committed:
refactor: 💡 do changes as request in PR#12

use of async/await instead of try/catch, add .env in .gitignore

✅ Closes: #40010

permil-successive committed:
refactor: 💡 resolve merge conflicts
